### PR TITLE
Api crude operations and screens of ApprovalEventRoles

### PR DESCRIPTION
**Pull request by karthikkharvi**
api crude operation and screen of eventRole

### CODE REVIEW

File: api/RS.API/RS.Data/Interfaces/IApprovalRepository.cs [hunk 1]
```diff
@@ -1,10 +1,15 @@
 ﻿using System.Collections.Generic;
 using RS.Entity.Models;
+using RS.ViewModel.User;
 
 namespace RS.Data.Interfaces
 {
     public interface IApprovalRepository : IRepository<Approvals>
     {
         List<ApprovalEvents> GetApprovalEvents(int approvalId);
+        List<ApprovalEvents> GetAllApprovalEvents();
```
Comment (salilsaini868) [inline api/RS.API/RS.Data/Interfaces/IApprovalRepository.cs]: filter result by approval id or remove this function

File: api/RS.API/RS.Data/Logic/ApprovalRepository.cs [hunk 2]
```diff
@@ -22,5 +23,29 @@ public List<ApprovalEvents> GetApprovalEvents(int approvalId)
                 .Where(t => t.Approval.ApprovalId == approvalId)
                 .OrderBy(t => t.ApprovalEventOrder).ToList();
         }
+
+        public List<ApprovalEvents> GetAllApprovalEvents()
+        {
+            return _context.ApprovalEvents.Include(t => t.ApprovalActions)
+                .OrderBy(t => t.ApprovalEventOrder).ToList();
+        }
```
Comment (salilsaini868) [inline api/RS.API/RS.Data/Logic/ApprovalRepository.cs]: remove this or add condition of approval id

File: api/RS.API/RS.Data/Logic/ApprovalRepository.cs [hunk 3]
```diff
@@ -22,5 +23,29 @@ public List<ApprovalEvents> GetApprovalEvents(int approvalId)
...
+        public void ManageApprovalEventRole(ApprovalEventRoles approvalEventRole)
+        {
+            _context.ApprovalEventRoles.Add(approvalEventRole);
+        }
+
+        public List<ApprovalEventRoles> GetAllApprovalEventRole()
+        {
+            return _context.ApprovalEventRoles.Include(t => t.ApprovalEvent).Include(s => s.Role).Include(r => r.User)
+                .Where(x => x.IsActive && !x.IsDeleted).ToList();
+        }
```
Comment (salilsaini868) [inline api/RS.API/RS.Data/Logic/ApprovalRepository.cs:36]: made check for approval

File: api/RS.API/RS.Data/Logic/ApprovalRepository.cs [hunk 4]
```diff
@@ -22,5 +23,29 @@ public List<ApprovalEvents> GetApprovalEvents(int approvalId)
...
+
+        public void UpdateUserEventRole(List<ApprovalEventRoles> approvalEventRoles)
+        {
+            approvalEventRoles.ForEach(x =>
+                _context.Entry(x).State = EntityState.Modified
+            );
```
Comment (salilsaini868) [inline api/RS.API/RS.Data/Logic/ApprovalRepository.cs]: ??

File: api/RS.API/RS.Data/Logic/UserRepository.cs [hunk 5]
```diff
@@ -29,6 +29,11 @@ public Users GetByID(Guid userId)
             return _context.Users.Include(t => t.UserRoles).ThenInclude(r => r.Role).FirstOrDefault(x => (x.IsActive && !x.IsDeleted) && (x.UserId == userId));
         }
 
+        public List<Users> GetUsersByRole(int roleId)
+        {
+            return _context.UserRoles.Where(x => x.RoleId == roleId).Select(x => x.user).ToList();
+        }
```
Comment (salilsaini868) [inline api/RS.API/RS.Data/Logic/UserRepository.cs]: add audit boolen checks

File: api/RS.API/RS.Service/Logic/ApprovalManagerService.cs [hunk 6]
```diff
@@ -57,5 +61,154 @@ public IResult GetApprovalEvents(int approvalId)
...
+                        eventViewModel.MapFromModel(t);
+                        if (t.ApprovalActions.Any())
+                        {
+                            var actionViewModel = new List<ApprovalActionViewModel>();
+                            eventViewModel.ApprovalActions = actionViewModel
+                                    .MapFromModel<ApprovalActions, ApprovalActionViewModel>(t.ApprovalActions.ToList());
+                        }
+                        return eventViewModel;
+                    }).ToList();
+                }
```
Comment (salilsaini868) [inline api/RS.API/RS.Service/Logic/ApprovalManagerService.cs]: remove this function

File: api/RS.API/RS.Service/Logic/ApprovalManagerService.cs [hunk 7]
```diff
@@ -57,5 +61,154 @@ public IResult GetApprovalEvents(int approvalId)
...
+            {
+                result.Message = e.Message;
+                result.Status = Status.Error;
+            }
+            return result;
+        }
+
+        void UpdateApprovalEventRole(ApprovalEventRoleViewModel approvalEventRoleViewModel)
+        {
+            var eventRoleList = _approvalRepository.GetAllApprovalEventRole().Where(x => x.ApprovalEventId == approvalEventRoleViewModel.ApprovalEventId && x.RoleId == approvalEventRoleViewModel.RoleId).ToList();
```
Comment (salilsaini868) [inline api/RS.API/RS.Service/Logic/ApprovalManagerService.cs:118]: use object from calling function

File: web/src/app/app.component.html [hunk 8]
```diff
@@ -7,8 +7,13 @@
         </div>
     </header>
 
+<<<<<<< HEAD
+<app-leftsidebar *ngIf="isTokenExist()"></app-leftsidebar>    
+<div class="content-wrapper"><router-outlet></router-outlet></div>
+=======
```
Comment (salilsaini868) [inline web/src/app/app.component.html]: remove conflics

**Review by salilsaini868 — CHANGES_REQUESTED** [review]

File: api/RS.API/RS.Data/Interfaces/IApprovalRepository.cs (hunk 1 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Data/Interfaces/IApprovalRepository.cs]: Done

File: api/RS.API/RS.Data/Logic/ApprovalRepository.cs (hunk 4 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Data/Logic/ApprovalRepository.cs]: Done

File: web/src/app/app.component.html (hunk 8 above)
Comment (karthikkharvi) [reply web/src/app/app.component.html]: Done

File: api/RS.API/RS.Data/Logic/UserRepository.cs (hunk 5 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Data/Logic/UserRepository.cs]: Done

File: api/RS.API/RS.Service/Logic/ApprovalManagerService.cs (hunk 7 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Service/Logic/ApprovalManagerService.cs:118]: Done

File: api/RS.API/RS.Service/Logic/ApprovalManagerService.cs (hunk 6 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Service/Logic/ApprovalManagerService.cs]: Done

File: api/RS.API/RS.Data/Logic/ApprovalRepository.cs (hunk 3 above)
Comment (karthikkharvi) [reply api/RS.API/RS.Data/Logic/ApprovalRepository.cs:36]: Done

**Review by salilsaini868 — APPROVED** [review]

**Pull request merged**